### DISCUSSION TITLE
Override the httpauth Unauthorized Error

### DISCUSSION THREAD

**Discussion by schroe-ja** [actix/actix-extras#427] [Q&A] [answered]
If I access my actix web API without a Bearer token then I want to override the `401`. Where can I do this?

I'm getting 
```
content-length: 0  
date: Fri,03 May 2024 05:15:38 GMT  
www-authenticate: Bearer 
```

But I want to create my own error message

```rust
impl HttpError {
    pub fn new(
        cause: Option<String>,
        message: Option<String>,
        status_code: String,
        error_type: ErrorResponse
    ) -> Self {
        Self {
            cause,
            message,
            status_code,
            error_type
        }
    }
}
```

**Comment by robjtede** [comment, marked as answer]
I updated one of the examples to show how optional auth works:
https://github.com/actix/actix-extras/blob/master/actix-web-httpauth/examples/middleware.rs

The main points are:
- use `HttpAuthentication::with_fn(validator)`
- have a validator signature that uses `credentials: Option<BearerAuth>` instead of `credentials: BearerAuth`.
